aten/src/ATen/native/sparse/SoftMax.cpp

```diff
@@ -407,6 +407,9 @@ void cpu_sparse_coo_softmax_backward(const Tensor& grad_input, const Tensor& gra
   indices.resize_as_(out_indices);
   indices.copy_(out_indices);
 
+  // TOOO: We need this when we call into from RedispatchBoxed (why?)
+  at::sparse::alias_into_sparse(grad_input, indices, values);
```

Comment:
Yeah, I don't really understand why this is necessary either. Would be good to understand better.

Reply:
The issue is that somewhere in the dispatcher a new TensorImpl is being created, making it so that resize_as_ on the returned _values does not affect the original sparse tensor.

So far I've found that TensorImpl returned by `_values` is TensorImpl returned by `op.call(self)` is different from that of `at::native::_values_sparse` (but only in certain situations), so the problem could be somewhere in the dispatcher.
 - Register in `native_functions.yaml` (no new TensorImpl)
 - Register in cpp extension (new TensorImpl) breaks
 - Register in `native_functions.yaml`, but call w/ redispatchBoxed, i.e., when we have this boxed fallback

I've tried a bit to bisect exactly where it occurs, but its surprisingly difficult to print trivial things in the dispatcher code: see https://github.com/pytorch/pytorch/pull/63652/files.

Reply:
cc @albanD

Reply:
Can you verify the that new code (in the boxed fallback) is equivalent to the code that would have been generated for sparse tensors? That seems like the most likely cause of the divergence. A good start is to paste the before code for this function.

Reply:
Good call, there was indeed a difference here. (AutoDispatchBelowADInplaceOrView <-> AutoDispatchBelowAutograd)
The issue is that we are dispatching into the InplaceOrView kernel for _values/_indices causing it to create a new TensorImpl in as_view. So a simple fix until we have our new InplaceOrView boxed kernel is have it AutoDispatchBelowADInplaceOrView always instead of AutoDispatchBelowAutograd.

Reply:
Great!

Reply:
So you're going to undo these changes now right? :)